packages/camera/camera_platform_interface/lib/src/platform_interface/camera_platform.dart

```diff
@@ -0,0 +1,115 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:plugin_platform_interface/plugin_platform_interface.dart';
+import 'package:camera_platform_interface/src/method_channel/method_channel_camera.dart';
+
+import '../../camera_platform_interface.dart';
+
+/// The interface that implementations of camera must implement.
+///
+/// Platform implementations should extend this class rather than implement it as `camera`
+/// does not consider newly added methods to be breaking changes. Extending this class
+/// (using `extends`) ensures that the subclass will get the default implementation, while
+/// platform implementations that `implements` this interface will be broken by newly added
+/// [CameraPlatform] methods.
+abstract class CameraPlatform extends PlatformInterface {
+  /// Constructs a CameraPlatform.
+  CameraPlatform() : super(token: _token);
+
+  static final Object _token = Object();
+
+  static CameraPlatform _instance = MethodChannelCamera();
+
+  /// The default instance of [CameraPlatform] to use.
+  ///
+  /// Defaults to [MethodChannelCamera].
+  static CameraPlatform get instance => _instance;
+
+  /// Platform-specific plugins should set this with their own platform-specific
+  /// class that extends [CameraPlatform] when they register themselves.
+  static set instance(CameraPlatform instance) {
+    PlatformInterface.verifyToken(instance, _token);
+    _instance = instance;
+  }
+
+  /// Completes with a list of available cameras.
+  Future<List<CameraDescription>> availableCameras() {
+    throw UnimplementedError('availableCameras() is not implemented.');
+  }
+
+  /// Initializes the camera on the device and returns its textureId.
+  Future<int> initializeCamera(
+    CameraDescription cameraDescription, {
+    ResolutionPreset resolutionPreset,
+    bool enableAudio,
+  }) {
+    throw UnimplementedError('initializeCamera() is not implemented.');
+  }
+
+  /// Returns a Stream of [CameraEvent]s.
+  Stream<CameraEvent> cameraEventsFor(int textureId) {
+    throw UnimplementedError('videoEventsFor() has not been implemented.');
+  }
+
+  /// Captures an image and saves it to [path].
+  Future<void> takePicture(int textureId, String path) {
```

Comment:
Personally I think you are right, I took a look at the iOS and Android code and it looks like we simply receive a byte array from the platform and the plugin it self writes it to disk.  I think makes sense to return this array to the application and let it worry about what to do with it (this will definitely fit a lot better with the web implementation).

@bparrishMines, @amirh how do you feel?

Reply:
The XFile allows to be initialized from "bytes", so there's no real need to write anything to file (until the user really wants to).

(I thought the native side would write to disk, and return the path, not that the whole data would be passed down through the platform interface)